app/services/csv/scheme_csv_service.rb

```diff
@@ -10,17 +10,22 @@ def initialize(download_type:)
     def prepare_csv(schemes)
       CSV.generate(headers: true) do |csv|
         csv << attributes
-        schemes.find_each do |scheme|
-          if @download_type == "schemes"
+
+        case @download_type
```

Comment:
I've refactored this to only check download type once, rather than on every loop, and also am using each rather than find_each (even though it's less efficient) because we wanted to preserve ordering. If it's too slow I'll revert this or consider ordering the csv after we've generated it.